src/plugins/core/borders.ts

```diff
@@ -105,53 +104,15 @@ export class BordersPlugin extends CorePlugin<BordersPluginState> implements Bor
         const elements = [...cmd.elements].sort((a, b) => b - a);
         for (const group of groupConsecutive(elements)) {
           if (cmd.dimension === "COL") {
-            if (group[0] >= this.getters.getNumberCols(cmd.sheetId)) {
-              for (let row: HeaderIndex = 0; row < this.getters.getNumberRows(cmd.sheetId); row++) {
-                this.history.update(
-                  "borders",
-                  cmd.sheetId,
-                  group[0] + 1,
-                  row,
-                  "vertical",
-                  undefined
-                );
-              }
-            }
-            if (group[group.length - 1] === 0) {
-              for (let row: HeaderIndex = 0; row < this.getters.getNumberRows(cmd.sheetId); row++) {
-                this.history.update("borders", cmd.sheetId, 0, row, "vertical", undefined);
-              }
-            }
-            const zone = this.getters.getColsZone(
-              cmd.sheetId,
-              group[group.length - 1] + 1,
-              group[0]
-            );
+            const start = Math.min(group[0], group[group.length - 1]);
+            const end = Math.max(group[0], group[group.length - 1]);
+            const zone = this.getters.getColsZone(cmd.sheetId, start, end);
             this.clearInsideBorders(cmd.sheetId, [zone]);
             this.shiftBordersHorizontally(cmd.sheetId, group[0] + 1, -group.length);
           } else {
-            if (group[0] >= this.getters.getNumberRows(cmd.sheetId)) {
-              for (let col = 0; col < this.getters.getNumberCols(cmd.sheetId); col++) {
-                this.history.update(
-                  "borders",
-                  cmd.sheetId,
-                  col,
-                  group[0] + 1,
-                  "horizontal",
-                  undefined
-                );
-              }
-            }
-            if (group[group.length - 1] === 0) {
-              for (let col = 0; col < this.getters.getNumberCols(cmd.sheetId); col++) {
-                this.history.update("borders", cmd.sheetId, col, 0, "horizontal", undefined);
-              }
-            }
-            const zone = this.getters.getRowsZone(
-              cmd.sheetId,
-              group[group.length - 1] + 1,
-              group[0]
-            );
+            const start = Math.min(group[0], group[group.length - 1]);
```

Comment:
Shouldn't we use this `start` value at line 117 instead of `group[0] + 1` ?

Reply:
(same comment for below)

Reply:
apparently I can't read, the elements were sorted in descending order on purpose on line 104 so there was no need to redefin start/end to begin with.